Automattic/wp-calypso
# Fix Jetpack Search Marketplace product purchase component

When the user purchased a Jetpack Search subscription from Settings > Performance, the Jetpack Search product displayed in the Marketplace appeared as inactive and it was possible to purchase it again.

#### Proposed Changes

* Fix the validation logic for the isPreinstalledPremiumPluginActive and PluginDetailsCTAPreinstalledPremiumPlugins component.

#### Testing Instructions



* Use the Calypso live link
* Currently there's a regression unrelated to this PR. To properly test it, apply this feature flag `flags=-marketplace-jetpack-plugin-search`
*  Get a new Jetpack site and connect it to your account
* Purchase a new Jetpack Search subscription from Settings > Performance
* Go to Plugins > Add new and search for the Jetpack Search product
* Notice that it appears as active and installed
* Click on the product details and make sure that you can't buy it again.
* 

#### Pre-merge Checklist






Related to #66111
